### ISSUE TITLE
C-style casts should not be flagged in C source files

### ISSUE THREAD

**Issue by ptomato** [cpplint/cpplint#190]
I have a project written mostly in C++, but some files have to be written in C and compiled by a C compiler. I get messages like:
```
./libgjs-private/gjs-util.c:143:  Using C-style cast.  Use reinterpret_cast<char*>(...) instead  [readability/casting] [4]
```
However, there's no alternative to a C-style cast in C. It seems like this lint message should not be given if the file name ends in `.c`.

**Comment by tkruse** [comment]
I don't know how much sense it makes to run a C++ linter on C files. It's not intended for that, even if it may have some limited useful output. I don't think the defaults of cpplint can be changed, given existing users might rely on files named .c being treated as C++.

**Comment by ptomato** [comment]
OK, thanks for the clear delineation of scope :smile: I guess this can be closed then.

**Comment by gringolito** [comment]
I understand the argument about running a C++ linter on C files and I don't totally disagree with it, anyway, the cpplint is often (even though not always) self-declared as `a command-line tool to check C/C++ files` and even the [.pre-commit-hooks.yaml](https://github.com/cpplint/cpplint/blob/develop/.pre-commit-hooks.yaml) file is configured to accept C files as input sources.

```yaml
- id: cpplint
  name: cpplint
  description: Static code analysis of C/C++ files
  language: python
  files: \.(h\+\+|h|hh|hxx|hpp|cuh|c|cc|cpp|cu|c\+\+|cxx|tpp|txx)$
  entry: cpplint
```

And I like to run the cpplint against C files because it provides me some useful outputs that applies broadly in C/C++. Just to mention a few that I've found running it on an old C project that I've started maintaining:
- `Consider using rand_r(...) instead of rand(...) for improved thread safety.  [runtime/threadsafe_fn] [2]`
- `Use int16/int64/etc, rather than the C type long  [runtime/int] [4]`
- `"bar.h" already included at foo.c:2  [build/include] [4]`
- `foo.c should include its header file foo.h  [build/include] [5]`
- `No #ifndef header guard found, suggested CPP variable is: FOO_BAR_BAZ_H_ [build/header_guard] [5]`
- `#ifndef header guard has wrong style, please use: FOO_BAR_BAZ_H_ [build/header_guard] [5]`
- `#endif line should be "#endif  // FOO_BAR_BAZ_H_"  [build/header_guard] [5]`
- `If an else has a brace on one side, it should have it on both  [readability/braces] [5]`

**Comment by gringolito** [comment]
Looking into the code, seem that this option should be suppressed for C files after all, so I really think that something is broken in the current implementation:

https://github.com/cpplint/cpplint/blob/2cba6ce8dfccfbe14dfe668c372149d9c39948a2/cpplint.py#L382-L385

**Comment by tkruse** [comment]
@gringolito : good point, I agree it looks like a bug that this suppression is not working

**Comment by JolonB** [comment]
I don't think this is a bug (unless it has been fixed recently). It looks like you need to add `// LINT_C_FILE`  to any files that you want to be linted as C.

**Comment by ptomato** [comment]
This seems to have been fixed.
